### ISSUE TITLE
[Fleet] User are not able to configure server fields from Fleet 

### ISSUE THREAD

**Issue by nchaulet** [elastic/fleet-server#2303]
## Description 

It seems that before 8.6 user were able to configure allowed ssl protocol using that config and it does not work anymore
<img width="600" alt="Screen Shot 2023-01-27 at 3 54 42 PM" src="https://user-images.githubusercontent.com/1336873/215185136-7a54b060-61aa-4885-9d6b-fbb82827e915.png">

when running nmap with `nmap -script ssl-enum-ciphers -p 8220 127.0.0.1` you can see that the tls protocol is not restricted
<img width="400" alt="Screen Shot 2023-01-27 at 3 57 05 PM" src="https://user-images.githubusercontent.com/1336873/215185381-55651496-7c81-4c27-9d74-7b3a07afe11e.png">

### Investigation details

The following config is generated in Fleet 
<img width="649" alt="Screen Shot 2023-01-27 at 3 54 59 PM" src="https://user-images.githubusercontent.com/1336873/215185086-1982015d-af74-4ac0-a7d2-13391faf3a76.png">

I tried to manual edit a policy and it seems that policy works well without the two `server` and `server.ssl.supported_protocol` 

```
id: fleet-server-fleet_server-8c1f2fbd-4e5b-450a-8565-2229c4437b81
revision: 3
name: fleet_server-1
type: fleet-server
data_stream:
  namespace: default
use_output: default
package_policy_id: 8c1f2fbd-4e5b-450a-8565-2229c4437b81
server:
  port: 8220
  host: 0.0.0.0
  ssl:
    supported_protocols:
    - TLSv1.2
meta:
  package:
    name: fleet_server
    version: 1.2.0

```

**Comment by rroemer** [comment]
I don't think this is only an issue for tls settings. 
I have configured `server.timeouts.checkin_long_poll: 90s` which is no longer working.

**Comment by michel-laterman** [comment]
@rroemer, I have a fix for the issue that will land in the 8.7 release.

I've added a backport label for 8.6, but there is no current plans for another 8.6 release
